mvn-resolver-transport-http3/src/test/java/com/artipie/aether/transport/http3/MavenResolverIT.java

```diff
@@ -111,39 +118,87 @@ public void testJettyLocalhostConnection() throws Exception {
         client.stop();
     }
 
+    @Test
+    public void testJettyLocalhostPut() throws Exception {
+        restartPutServer();
+        final HTTP3Client h3Client = new HTTP3Client();
+        HttpClientTransportOverHTTP3 transport = new HttpClientTransportOverHTTP3(h3Client);
+        final HttpClient client = new HttpClient(transport);
+        client.start();
+        h3Client.getClientConnector().getSslContextFactory().setTrustAll(true);
+        final String testData = "test data";
+        final byte[] bytes = testData.getBytes(StandardCharsets.UTF_8);
+        final ContentResponse response = client.newRequest("https://localhost:7445/test1").
+            method(HttpMethod.PUT).body(
+                new InputStreamRequestContent(new ByteArrayInputStream(bytes))
+            ).headers(httpFields -> httpFields.add(HttpHeader.CONTENT_LENGTH, bytes.length)).send();
+        assertEquals(200, response.getStatus());
+        final Container.ExecResult result = caddyProxy.execInContainer("cat /srv/test1".split(" "));
+        assertEquals(0, result.getExitCode());
+        assertEquals(testData, result.getStdout());
+        client.stop();
+    }
+
+    @Test
+    public void testTransporterPutAnon() throws Exception {
+        final String repo = "https://localhost:7445/";
+        restartPutServer();
+        final HttpTransporterFactory factory = new HttpTransporterFactory();
+        final PutTask task = new PutTask(URI.create("test1")).setListener(new TransportListener() {});
+        final String testData = "test data";
+        task.setDataString(testData);
```

Comment:
Maybe would it be better to use binary test data? 
WDYT?

Reply:
Good point, I'll switch to binary test data.